packages/mds-transactions-service/repository/index.ts

```diff
@@ -58,15 +60,27 @@ class TransactionReadWriteRepository extends ReadWriteRepository {
       return {}
     }
     function who(): { provider_id?: UUID } {
-      if (provider_id) {
-        return { provider_id }
-      }
-      return {}
+      return provider_id ? { provider_id } : {}
     }
     try {
       const connection = await connect('ro')
-      const entities = await connection.getRepository(TransactionEntity).find({ where: { ...who(), ...when() } })
-      return entities.map(TransactionEntityToDomain.mapper())
+      const queryBuilder = connection
+        .getRepository(TransactionEntity)
+        .createQueryBuilder('transactionentity') // yuk!
+        .where({ ...who(), ...when() })
+      const query: PagingQuery = {
+        limit: 10,
+        order: 'ASC'
+      }
+      // probably cleaner way to do this? not sure.
+      // also i'm tolerating both before and after and using just after in that case.
+      if (after) {
+        query.afterCursor = after
+      } else if (before) {
+        query.beforeCursor = before
+      }
+      const { data, cursor } = await buildPaginator({ entity: TransactionEntity, query }).paginate(queryBuilder)
```

Comment:
You _could_ remove the local/if/then/else and keep the paging options together as follows. Your preference; just an alternative:
```suggestion
      const { data, cursor } = await buildPaginator({
        entity: TransactionEntity,
        query: {
          limit: 10,
          order: 'ASC',
          afterCursor: after,
          beforeCursor: after ? undefined : before
        }
      }).paginate(queryBuilder)
```

Reply:
so i was reacting to an issue i'd discovered earlier in the where() clause.  i'll call you to discuss, takes too long to type.